### PR TITLE
Validate ANSIBLE_ injectors on save and increase verbosity

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
Fixes issue https://github.com/ansible/awx/issues/2359, also fixes at least one other issue, which I will link up after the PR is open.

##### ISSUE TYPE
 - Feature Pull Request
 - Bugfix Pull Request

##### COMPONENT NAME
 - API

##### AWX VERSION
```
1.0.8-23
```

##### ADDITIONAL INFORMATION
old behavior:

Saved credential types when ANSIBLE_ env vars were given, and silently declined to inject them to the job runtime environment. I know this is documented behavior, but realistically, it will take an hour of someone banging their head against a wall, and _then_ they will find the documentation. If you experience the pain of writing injectors this way, it feels like starkly user-hostile behavior.

new behavior:

The validation is reused in two different places. The field validation method covers saving the credential type. Here are some screen caps.

screen shot 2018-10-03 at 8 56 00 am

screen shot 2018-10-03 at 9 07 21 am

The other place is the old place where we checked this, when running a job. Now we will show a log message explaining that something didn't go right. Job still runs.

```
awx_1        | 12:39:45 celeryd.1   | 2018-10-03 12:39:45,370 ERROR    awx.main.models.credential Ignoring prohibited env var ANSIBLE_FOO, reason: [u'env var ANSIBLE_FOO may affect Ansible configuration so its use is not allowed in credentials']
```

I could see _maybe a little risk_ allowing old credential types to linger here, while at the same time having this validation in the field definition. The concern might be that if the _system_ edits a custom credential type, this could throw a validation error. I don't know if this is a very realistic concern, but if it is, we can keep the method in the field, but not call it in the field validate, and have the serializer manually grab the field and call the method. That would accomplish the same goal. Probably not necessary, but I'll make that change if requested.

### CODE REVIEW

File: awx/main/tests/functional/api/test_credential_type.py [hunk 1]
```diff
@@ -359,18 +359,17 @@ def test_create_with_valid_injectors(get, post, admin):
         },
         'injectors': {
             'env': {
-                'ANSIBLE_MY_CLOUD_TOKEN': '{{api_token}}'
+                'AWX_MY_CLOUD_TOKEN': '{{api_token}}'
```
Comment (AlanCoding) [inline awx/main/tests/functional/api/test_credential_type.py:362]: Our own tests were saving disallowed injector configurations.

File: awx/main/fields.py [hunk 2]
```diff
@@ -783,6 +788,19 @@ def schema(self, model_instance):
             'additionalProperties': False
         }
 
+    def validate_env_var_allowed(self, env_var):
+        if env_var.startswith('ANSIBLE_'):
+            raise django_exceptions.ValidationError(
+                _('env var {} may affect Ansible configuration so its '
```
Comment (ryanpetrello) [inline awx/main/fields.py]: Let's make this say `environment variable {}`

**Review by ryanpetrello — APPROVED** [review]

**Comment by AlanCoding** [comment]
Thanks for the review! I looked back over this and I feel confident for a merge in its current state.

**Comment by ryanpetrello** [comment]
recheck

**Pull request merged**